scan/lib/scan/test_command_generator.rb

```diff
@@ -31,8 +31,9 @@ def options
 
         options = []
         options += project_path_array
-        options << "-sdk '#{config[:sdk]}'" if config[:sdk]
-        options << destination # generated in `detect_values`
+        options << "-sdk '#{config[:sdk]}'" if config[:sdk] && !config[:build_for_testing]
+        options << destination unless config[:build_for_testing] # generated in `detect_values`
+        options << "  -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO" if config[:build_for_testing]
```

Comment:
I think we still need a destination here right? the idea is everything should be the same as if you called `test`

See http://devstreaming.apple.com/videos/wwdc/2016/409jh83sf1h8dqrt00q/409/409_advanced_testing_and_continuous_integration.pdf